subsys/sd/sdmmc.c

```diff
@@ -1172,7 +1172,7 @@ static int sdmmc_read_status(struct sd_card *card)
 /* Waits for SD card to be ready for data. Returns 0 if card is ready */
 static int sdmmc_wait_ready(struct sd_card *card)
 {
-	int ret, timeout = CONFIG_SD_DATA_TIMEOUT;
+	int ret, timeout = CONFIG_SD_DATA_TIMEOUT * 1000;
```

Comment:
(question) is it ok that you reduced maximum data timeout?
before this change timeout may be `2^31 - 1` ms, i.e. ~24,85 days
after this change we can wait 35,8 minutes

I see that we have a mistake here, but maybe it is better to use 64-bit type
what if you have low speed SD/MMC and write/read a lot of blocks of data, e.g. a few gigabytes?